src/bun.js/node/types.zig

```diff
@@ -2848,20 +2865,35 @@ pub const Path = struct {
         lenSum += 1;
 
         const pathObject = .{ .root = root, .dir = dir, .base = base, .ext = ext, .name = _name };
-        const bufLen = @max(lenSum, MAX_PATH_BYTES);
-        if (bufLen > MAX_PATH_BYTES) {
+        const size = PathSize(u8);
+        const bufLen = getPathLenT(u8, lenSum);
+        if (bufLen > size.MAX) {
             const buf = allocator.alloc(u8, bufLen) catch bun.outOfMemory();
             defer allocator.free(buf);
             return if (isWindows)
                 toJSString(globalObject, formatWindow(pathObject, buf))
             else
                 toJSString(globalObject, formatPosix(pathObject, buf));
         }
-        var buf: PathBuffer = undefined;
+        const buf = switch (bufLen) {
```

Comment:
since bufLen is not known at compile time, all 3 branches are present in the binary, meaning all stack allocations in there are present. so all three of these buffers are present

Reply:
Oh that's super lame. Will fork it.

Reply:
> so all three of these buffers are present

Fixed this.